package-dev/Plugins/macOS/SentryNativeBridge.m

```diff
@@ -0,0 +1,220 @@
+                  forKey:@"timestamp"];
+        }
+
+        if (message != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:message] forKey:@"message"];
+        }
+
+        if (type != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:type] forKey:@"type"];
+        }
+
+        if (category != NULL) {
+            [breadcrumb setValue:[NSString stringWithUTF8String:category] forKey:@"category"];
+        }
+
+        [breadcrumb setValue:[NSNumber numberWithInt:level] forKey:@"level"];
+
+        [scope performSelector:@selector(addBreadcrumb:) withObject:breadcrumb];
+    });
+}
+
+void SentryNativeBridgeSetExtra(const char *key, const char *value)
+{
+    if (key == NULL) {
+        return;
+    }
+
+    SentryConfigureScope(^(id scope) {
+        if (value != NULL) {
+            [scope performSelector:@selector(setExtraValue:forKey:)
+                        withObject:[NSString stringWithUTF8String:value]
+                        withObject:[NSString stringWithUTF8String:key]];
+        } else {
+            [scope performSelector:@selector(removeExtraForKey:)
+                        withObject:[NSString stringWithUTF8String:key]];
+        }
+    });
+}
+
+void SentryNativeBridgeSetTag(const char *key, const char *value)
+{
+    if (key == NULL) {
+        return;
+    }
+
+    SentryConfigureScope(^(id scope) {
+        if (value != NULL) {
+            [scope performSelector:@selector(setTagValue:forKey:)
+                        withObject:[NSString stringWithUTF8String:value]
+                        withObject:[NSString stringWithUTF8String:key]];
+        } else {
+            [scope performSelector:@selector(removeTagForKey:)
+                        withObject:[NSString stringWithUTF8String:key]];
+        }
+    });
+}
+
+void SentryNativeBridgeUnsetTag(const char *key)
+{
+    if (key == NULL) {
+        return;
+    }
+
+    SentryConfigureScope(^(id scope) {
+        [scope performSelector:@selector(removeTagForKey:)
+                    withObject:[NSString stringWithUTF8String:key]];
+    });
+}
+
+void SentryNativeBridgeSetUser(
+    const char *email, const char *userId, const char *ipAddress, const char *username)
+{
+    if (email == NULL && userId == NULL && ipAddress == NULL && username == NULL) {
+        return;
+    }
+
+    SentryConfigureScope(^(id scope) {
+        id user = [[SentryUser alloc] init];
+
+        if (email != NULL) {
+            [user setValue:[NSString stringWithUTF8String:email] forKey:@"email"];
+        }
+
+        if (userId != NULL) {
+            [user setValue:[NSString stringWithUTF8String:userId] forKey:@"userId"];
+        }
+
+        if (ipAddress != NULL) {
+            [user setValue:[NSString stringWithUTF8String:ipAddress] forKey:@"ipAddress"];
+        }
+
+        if (username != NULL) {
+            [user setValue:[NSString stringWithUTF8String:username] forKey:@"username"];
+        }
+
+        [scope setUser:user];
```

Comment:
`m`: I'm a bit confused. Above we use `performSelector` and here we can call the method directly. Why is that?

Reply:
it gives a lot of warnings unless the argument is nil or `id`:

`instance method '-setExtraValue:forKey:' not found (return type defaults to 'id') [-Wobjc-method-access]`

with some Unity version, these warnings actually caused the build to fail.

I'll also change setUser to performSelector and add this comment to the code.